### PR TITLE
Get inventory dicts

### PR DESCRIPTION
**Pull request by brandomando**
Adding functions to nornir.core.inventory.Inventory to abstract serialization of the running inventory into dictionaries. Functions for: full inventory, defaults, groups, and hosts. This makes the inventory available in an exportable format that users could then build plugins to push out into various formats.

A simple future usecase would be to leverage the dicts to export to simple inventory files.  This would allow dynamically added inventory items to be captured and placed into a more permanent inventory.

### CODE REVIEW

File: nornir/core/inventory.py [hunk 1]
```diff
@@ -460,3 +460,27 @@ def add_group(self, name: str, **kwargs) -> None:
...
+        """
+        Return serialized dictionary of inventory
+        """
+        return deserializer.inventory.Inventory.serialize(self).dict()
+
+    def get_defaults_dict(self) -> Dict:
+        """
+        Returns serialized dictionary of defaults from inventory
+        """
+        return self.get_inventory_dict().get("defaults")
```
Comment (dbarrosop) [inline nornir/core/inventory.py]: I think we should call instead:

```
         return deserializer.inventory.Defaults.serialize(self.defaults).dict()
```

File: nornir/core/inventory.py [hunk 2]
```diff
@@ -460,3 +460,27 @@ def add_group(self, name: str, **kwargs) -> None:
...
+
+    def get_groups_dict(self) -> Dict:
+        """
+        Returns serialized dictionary of groups from inventory
+        """
+        return self.get_inventory_dict().get("groups")
```
Comment (dbarrosop) [inline nornir/core/inventory.py]: I think we should call instead:

```
         return {k: deserializer.inventory.InventoryElement.serialize(v).dict() for k, v in self.groups}
```

File: nornir/core/inventory.py [hunk 3]
```diff
@@ -460,3 +460,27 @@ def add_group(self, name: str, **kwargs) -> None:
...
+
+    def get_hosts_dict(self) -> Dict:
+        """
+        Returns serialized dictionary of hosts from inventory
+        """
+        return self.get_inventory_dict().get("hosts")
```
Comment (dbarrosop) [inline nornir/core/inventory.py]: I think we should call instead:

```
         return {k: deserializer.inventory.InventoryElement.serialize(v).dict() for k, v in self.hosts}
```

**Comment by dbarrosop** [comment]
Looks good, just a few minor comments to make the code more efficient.

File: nornir/core/inventory.py (hunk 3 above)
Comment (brandomando) [reply nornir/core/inventory.py]: I see what you're saying on this and the groups. It avoids calling an additional function to perform the action. One minor correction however, just adding `.items()` on both this and groups
```
        return {k: deserializer.inventory.InventoryElement.serialize(v).dict() for k, v in self.hosts.items()}
```

**Comment by brandomando** [comment]
@dbarrosop , I believe I have addressed your changes. Let me know if there's anything else.

Thanks

**Comment by dbarrosop** [comment]
Awesome!

**Pull request merged**